dhall/tests/Dhall/Test/QuickCheck.hs

```diff
@@ -171,7 +171,13 @@ instance Arbitrary Header where
                   , ( 1, arbitrary)
                   ]
 
-          commentText = Text.pack <$> Test.QuickCheck.listOf commentChar
+          noInteriorBlockComments text =
+              not (Text.isInfixOf "{-" text || Text.isInfixOf "-}" text)
+
+          commentText =
+              suchThat
+                  (Text.pack <$> Test.QuickCheck.listOf commentChar)
+                  noInteriorBlockComments
```

Comment:
What's the motivation for filtering `{-` and `-}`?

Reply:
@sjakobi: They also lead to malformed header comments.  For example: `{- abc -} def -}`